pages/installation.md

```diff
@@ -8,25 +8,302 @@ permalink: /installation/
+
+# Endpoints HTTP
+OOP_ENDPOINTS_HTTP_OUTPUT_Q=oop.hasauth.relay
+OOP_REQUEST_TIMEOUT=1000
+OOP_ENDPOINTS_HTTP_MAX_RETRIES=3
+
+# Relay
+OOP_RELAY_INPUT_Q=oop.hasauth.relay
+OOP_RECURSIVE_TEMPR_Q=oop.hasauth.temprs
+OOP_CORE_RESPONSE_Q=oop.core.transmissions
+
+# Core
+OOP_AMQP_ADDRESS=amqp://localhost
+OOP_RENDERER_PATH=/path/to/renderer/
+
+OOP_CORE_SCHEME=http://
+OOP_CORE_PORT=9001
+OOP_CORE_PATH=/
+
+PORT=9001
+
+OOP_CORE_INTERFACE_SCHEME=http://
+OOP_CORE_INTERFACE_PORT=80
+OOP_CORE_INTERFACE_PATH=/
+OOP_CORE_FROM_ADDRESS=example@mail.co.uk
+
+# Mail setup
+OOP_CORE_FROM_ADDRESS=
+OOP_CORE_SMTP_ADDRESS=
+OOP_CORE_SMTP_PORT=25
+OOP_CORE_SMTP_DOMAIN=
+OOP_CORE_SMTP_USER_NAME=
+OOP_CORE_SMTP_PASSWORD=
+OOP_CORE_SMTP_AUTHENTICATION=
+OOP_CORE_SMTP_ENABLE_STARTTLS_AUTO=
+
+DATABASE_HOST=localhost
+POSTGRES_USER=postgres
+POSTGRES_PASSWORD=password
+POSTGRES_DB=oop_core_development
+
+SECRET_KEY_BASE={SECRETKEY}
+```
+
+### oop-core
+
+Clone the application `git clone http://github.com/open-interop/oop-core.git`
+In the app directory install all dependencies using  `bundle install`
+Create a symlink to the .env file created earlier - `ln -nfs /path/to/.env /path/to/oop-core/`
+Ensure all variables are set up correctly in the config directory:
+
+###### `database.yml`
+
+```
+# config/database.yml
+
+default: &default
+  adapter: postgresql
+  encoding: unicode
+  pool: <%= ENV.fetch("RAILS_MAX_THREADS") { 5 } %>
+  host: <%= ENV.fetch('DATABASE_HOST') %>
+  username: <%= ENV.fetch('POSTGRES_USER') %>
+  password: <%= ENV.fetch('POSTGRES_PASSWORD') %>
+  database: <%= ENV.fetch('POSTGRES_DB') %>
+  variables:
+    statement_timeout: 5000
+
+production:
+  <<: *default
+```
+
+###### `storage.yml`
+
+```
+# config/storage.yml
+
+local:
+  service: Disk
+  root: <%= Rails.root.join("storage") %>
+```
+
+###### `secrets.yml`
+
+```
+# config/secrets.yml
+
+production:
+  secret_key_base: <%= ENV["SECRET_KEY_BASE"] %>
+```
+
+
+Create the database - `bundle exec rails db:create`
+Run the database migrations - `bundle exec rails db:migrate`
+
+For the initial setup run - `bin/rails open_interop:setup_initial_account`
+
+To start core run - `bin/rails server -b HOSTNAME -p 9001` where HOSTNAME is the host core is accessible from - this should be reflected in the `.env` file above
+
+`oop-core` includes the transmission queue, to start this run - `bin/rails runner TransmissionQueue.retrieve_transmissions`
+
+### Microservices:
+
+The following services can all be installed and set up in exactly the same way, cloning the repo and using yarn to install any dependencies. 
+
+#### oop-gateway
+
+`git clone http://github.com/open-interop/oop-gateway.git`
+
+`cd oop-gateway`
+
+`ln -nfs /path/to/.env /path/to/oop-gateway`
+
+`yarn install`
+
+When all microservices are installed and core is up, run `yarn start`
+
+#### oop-tempr
+
+`git clone http://github.com/open-interop/oop-tempr.git`
+
+`cd oop-tempr`
+
+`ln -nfs /path/to/.env /path/to/oop-tempr`
```

Comment:
Maybe something like this would make more sense?

`ln -nfs /path/to/.global-env /path/to/oop-tempr/.env`